### PR TITLE
[ADD] stock_reception_screen_measuring_device

### PR DESCRIPTION
**Pull request by gurneyalex**
this is a glue module between stock_reception_screen and stock_measuring_device

Depends on:
* https://github.com/OCA/stock-logistics-warehouse/pull/1121

### CODE REVIEW

File: stock_reception_screen_measuring_device/views/measuring_device_view.xml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_zippcube_device_form" model="ir.ui.view">
+        <field name="name">zippcube.device.form</field>
+        <field name="model">zippcube.device</field>
```
Comment (simahawk) [inline stock_reception_screen_measuring_device/views/measuring_device_view.xml]: this depends on zippcube module, not on the generic one

File: stock_reception_screen_measuring_device/models/measuring_device.py [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class MeasuringDevice(models.Model):
+    _inherit = "measuring.device"
+
+    is_default = fields.Boolean(
```
Comment (simahawk) [inline stock_reception_screen_measuring_device/models/measuring_device.py:10]: maybe can be moved to the base module?

**Review by simahawk — CHANGES_REQUESTED** [review]

File: stock_reception_screen_measuring_device/models/measuring_device.py (hunk 2 above)
Comment (gurneyalex) [reply stock_reception_screen_measuring_device/models/measuring_device.py:10]: we don't need the concept in the base module -> I'd rather keep it here, at least for now.

File: stock_reception_screen_measuring_device/models/stock_reception_screen.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    def measure_current_packaging(self):
+        self.ensure_one()
+        device = self.env["measuring.device"].search(
+            [("is_default", "=", True)], limit=1
+        )
+        if not device:
+            error_msg = _("No default device set, please configure one.")
+            _logger.error(error_msg)
+            self._notify(error_msg)
+            return UserError(error_msg)
```
Comment (mmequignon) [inline stock_reception_screen_measuring_device/models/stock_reception_screen.py:61]: ~~No fallback to a non-default device?~~

File: stock_reception_screen_measuring_device/readme/USAGE.rst [hunk 4]
```diff
@@ -0,0 +1,6 @@
+In the reception screen, click on the "Measure Current Packaging" button
+(visible if the packaging of the reception is set and has not
+weight/dimensions), and use the measuring device. The values should be directly
+updated on the packaging.
+
+Known limitation: this only works on the measuring device flagges as default.
```
Comment (mmequignon) [inline stock_reception_screen_measuring_device/readme/USAGE.rst]:
```suggestion
Known limitation: this only works on the measuring device flagged as default. 
```
I have my answer for my previous comment.

**Review by mmequignon — APPROVED** [review]

File: stock_reception_screen_measuring_device/views/measuring_device_view.xml [hunk 5]
```diff
@@ -0,0 +1,29 @@
+<?xml version="1.0" ?>
+<odoo>
+    <record id="view_measuring_device_form" model="ir.ui.view">
+        <field name="name">measuring.device.form</field>
+        <field name="model">measuring.device</field>
+        <field
+            name="inherit_id"
+            ref="stock_measuring_device.measuring_device_form"
```
Comment (mmequignon) [inline stock_reception_screen_measuring_device/views/measuring_device_view.xml]:
```suggestion
            ref="stock_measuring_device.view_measuring_device_form"
```

**Review by simahawk — APPROVED** [review]

**Comment by mmequignon** [comment]
@sbidoul Hello, we have an issue here.
I squashed the commits before the merge, but the bot merged the commits as they were before the squash.
Do you have any clue about what is doable to revert / fix this?
Thank you!

**Comment by mmequignon** [comment]
> @sbidoul Hello, we have an issue here.
> I squashed the commits before the merge, but the bot merged the commits as they were before the squash.
> Do you have any clue about what is doable to revert / fix this?
> Thank you!

It has also losts the last modifications in the process:
On the PR : https://github.com/OCA/wms/pull/214/commits/964185de086132cc004fba96ec353b069f687c19#diff-a480c541cdff2026960eb234b46ca3b434bf93b0dd3bb228a35642f9dbedd275R68

What has been merged : https://github.com/OCA/wms/blob/13.0/stock_reception_screen_measuring_device/models/stock_reception_screen.py#L68

**Comment by sbidoul** [comment]
@mmequignon @simahawk 

> I squashed the commits before the merge

Before, before, c'est vite dit :)

image

Could it be that GitHub sends webhook events in the past ?

> what is doable to revert / fix this?

A force push on the main branch (preferably not, because it requires a manual repair in weblate but exceptionnally, just for you we can do it ;), or some reverts.

**Comment by simahawk** [comment]
holy hack, I've pushed the btn too early! :smile: 

No because his last force-push was 2h ago

image

Likely only w/ few minutes of difference.

But that seems weird anyway because the branch has been created after (well, we cannot check anymore as it got deleted)

Maybe somehow a reference to some old parent commit remained and this messed up the checkout :man_shrugging: 


Well, no real harm if not a crapy changelog, like:

image

@mmequignon will open a PR w/ fixes... we fast track and that's it.

Thank you anyway!

**Comment by mmequignon** [comment]
@sbidoul Thank you!

**Pull request closed without merging**